adapters/repos/db/vector/hnsw/index.go

```diff
@@ -52,7 +52,7 @@ type hnsw struct {
 	// this is a point on the highest level, if we insert a new point with a
 	// higher level it will become the new entry point. Note tat the level of
 	// this point is always currentMaximumLayer
-	entryPointID int
+	entryPointID int64
```

Comment:
whole file: see above, all node ids should be represented by `uint64`, removing the need for the conversions.